src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx

```diff
@@ -264,30 +293,23 @@ export default function ManageMappingsPage(props: IManageMappingsPageProps) {
     }
   }
 
-  function UpdateExistingMapping(jsonValue, rows, idx, itemId) {
-    let exitEntry = false;
-    for (let i = 0; i < jsonValue.length; i++) {
-      if (
-        jsonValue[i].SharePointContentType == rows[idx].SharePointContentType
-      ) {
-        if (jsonValue[i].id == rows[idx].id) {
-          exitEntry = false;
-          break;
-        } else {
-          exitEntry = true;
-          break;
-        }
-      }
-    }
-    if (exitEntry == false) {
-      for (let j = 0; j < jsonValue.length; j++) {
-        if (jsonValue[j].id == rows[idx].id) {
-          jsonValue[j].SharePointContentType = rows[idx].SharePointContentType;
-          jsonValue[j].LaserficheContentType = rows[idx].LaserficheContentType;
-          jsonValue[j].toggle = !rows[idx].toggle;
-          break;
-        }
-      }
+  async function updateExistingMappingAsync(
+    jsonValue: ProfileMappingConfiguration[],
+    rows: ProfileMappingConfiguration[],
+    idx: number,
+    itemId: string
+  ) {
+    const spContentTypeMatch = jsonValue.find(
+      (mapping) =>
+        mapping.SharePointContentType === rows[idx].SharePointContentType
+    );
+    if (!spContentTypeMatch || spContentTypeMatch.id === rows[idx].id) {
```

Comment:
The logic here is a bit hard to read. maybe make `!spContentTypeMatch || spContentTypeMatch.id === rows[idx].id` a value and give it a good name.